Packages/PackageSources/meridian-simcache/html_ui/InGamePanels/SimCache/SimCache.js

```diff
@@ -48,13 +28,24 @@ class SimCachePanel extends UIElement {
     }
 
     onSubsystemsInitialized() {
+        this.CommBusListener.on("SimCache.TrackedCacheDataUpdatedEvent", this.onTrackedCacheDataUpdatedEvent.bind(this));
         this.CommBusListener.on("SimCache.CacheFoundEvent", this.onCacheFoundEvent.bind(this));
         this.CommBusListener.callWasm("SimCache.TrackerLoadedEvent", "");
     }
 
+    onTrackedCacheDataUpdatedEvent(eventData) {
+        if (this.m_timeoutID !== null) {
+            this.cancelPendingClose();
+        }
+        const data = JSON.parse(eventData);
+        this.m_titleElement.innerText = data.title;
+        this.updateRange(data.distance_meters);
+        this.m_containerElement.classList.remove("display-none");
```

Comment:
There are a few things in this file that make me think "what does that do?" 😛 . Worth adding comments or writing descriptive functions for things like this?

Reply:
Coming up in the next commit! :D